lib/containers/__tests__/home-tests.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+
+'use strict'
+import { ListViewDataSource } from 'react-native'
+import React from 'react'
+import renderer from 'react-test-renderer'
+
+import Home from '../home'
+
+jest.mock('../../components/home/artist_rails/artist_rail', () => 'ArtistRail')
+jest.mock('../../components/home/artwork_rails/artwork_rail', () => 'ArtworkRail')
+jest.mock('../../components/spinner.js', () => 'ARSpinner')
+jest.mock('../../components/opaque_image_view.js', () => 'AROpaqueImageView')
+
+describe('upon initialization', () => {
+  it('starts with isRefreshing set to false', () => {
+    const homeComponent = new Home(homeProps())
+    expect(homeComponent.state.isRefreshing).toBeFalsy()
+  })
+
+  it('creates a data source with a search bar and hero unit', () => {
+    const homeComponent = new Home(homeProps())
+    const dataSourceBlob = extractListViewDataBlob(homeComponent)
+    const searchBar = { type: 'search_bar', data: null }
+    const heroUnits = { type: 'hero_units', data: [] }
+    expect(dataSourceBlob).toContainEqual(searchBar)
+    expect(dataSourceBlob).toContainEqual(heroUnits)
+  })
+
+  it('orders modules correctly', () => {
+    const homeComponent = new Home(homeProps(3,3,3))
+    const dataSourceBlob = extractListViewDataBlob(homeComponent)
+    // order should be search bar, hero units, 2 artwork modules, one artist module
+    expect(dataSourceBlob[0].type).toBe('search_bar')
+    expect(dataSourceBlob[1].type).toBe('hero_units')
+    expect(dataSourceBlob[2].type).toBe('artwork')
+    expect(dataSourceBlob[3].type).toBe('artwork')
+    expect(dataSourceBlob[4].type).toBe('artist')
+  })
+})
+
+describe('layout', () => {
+  it('renders as expected', () => {
+    const homeComponent = renderer.create(<Home home={homeProps(3,3,3).home}/>).toJSON()
+    expect(homeComponent).toMatchSnapshot()
+  })
+})
+
+var homeProps = (nHeroUnits?: number, nArtistModules?: number, nArtworkModules?: number) => {
+  return {
+      home: {
+        hero_units: nHeroUnits ? modulesOfType('hero_units', nHeroUnits) : [],
```

Comment:
As we know this is a flow issue, it's best to use the flow ignore error syntax to skip the error:

```js
// $FlowFixMe: This is a bug with Flow + React-Native
import { ListViewDataSource } 
```